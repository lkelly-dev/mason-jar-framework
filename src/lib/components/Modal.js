import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'emotion';
import { Button } from './Button';
import { debugAttrs, debugMode } from '../utils/debugging';

const Modal = props => {
  const { context, elementId, children, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c)
  return (
    <div className="modal" tabIndex="-1" role="dialog" aria-hidden="true" id={elementId}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      <div className="modal-dialog" role="document">
        {children}
      </div>
    </div>
  );
};

const ModalTrigger = props => {
  const { context, children, elementId, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c);
  return (
    <Button data-toggle="modal" data-target={`#${elementId}`}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </Button>
  );
};

const ModalContent = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c);
  const modalContentClass = cx('modal-content', className);
  return (
    <div className={modalContentClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const ModalHeader = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c);
  const modalHeaderClass = cx('modal-header', className);
  return (
    <div className={modalHeaderClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const ModalBody = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c);
  const modalBodyClass = cx('modal-body', className);
  return (
    <div className={modalBodyClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const ModalFooter = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c);
  const modalFooterClass = cx('modal-footer', className);
  return (
    <div className={modalFooterClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </div>
  );
};

const ModalTitle = props => {
  const { context, children, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c);
  const modalTitleClass = cx('modal-title', className);
  return (
    <h3 className={modalTitleClass}
      {...(debug) ? debugAttrs(props): null}
      {...rest}>
      {children}
    </h3>
  );
};

const ModalClose = props => {
  const { context, className, ...rest } = props;
  const c =  (context) ? useContext(context) : null;
  const debug = debugMode(c);
  const modalCloseClass = cx('close', className);
  return (
    <button
      type="button"
      className={modalCloseClass}
      data-dismiss="modal"
      aria-label="Close"
      {...(debug) ? debugAttrs(props): null}
      {...rest}
    >
      <span aria-hidden="true">&times;</span>
    </button>
  );
};

Modal.Content = ModalContent;
Modal.Header = ModalHeader;
Modal.Body = ModalBody;
Modal.Footer = ModalFooter;
Modal.Title = ModalTitle;
Modal.Close = ModalClose;
Modal.Trigger = ModalTrigger;

export { Modal, ModalHeader, ModalTitle, ModalBody, ModalFooter, ModalClose, ModalTrigger };

Modal.propTypes = {
  elementId: PropTypes.string.isRequired,
};

ModalTrigger.propTypes = {
  elementId: PropTypes.string.isRequired,
};
