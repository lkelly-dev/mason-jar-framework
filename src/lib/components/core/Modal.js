import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'emotion';
import { Button } from './Button';

const Modal = props => {
  const { element, children } = props;
  // FIXME element is currently a required prop
  // FIXME is element actually the correct name? why not use the name id? and whats it for? etc
  return (
    <div className="modal" tabIndex="-1" role="dialog" aria-hidden="true" id={element}>
      <div className="modal-dialog" role="document">
        {children}
      </div>
    </div>
  );
};

const ModalTrigger = props => {
  const { children, element, ...rest } = props;
  return (
    <Button data-toggle="modal" data-target={`#${element}`} {...rest}>
      {children}
    </Button>
  );
};

const ModalContent = props => {
  const { children, className, ...rest } = props;
  const modalContentClass = cx('modal-content', className);
  return (
    <div className={modalContentClass} {...rest}>
      {children}
    </div>
  );
};

const ModalHeader = props => {
  const { children, className, ...rest } = props;
  const modalHeaderClass = cx('modal-header', className);
  return (
    <div className={modalHeaderClass} {...rest}>
      {children}
    </div>
  );
};

const ModalBody = props => {
  const { children, className, ...rest } = props;
  const modalBodyClass = cx('modal-body', className);
  return (
    <div className={modalBodyClass} {...rest}>
      {children}
    </div>
  );
};

const ModalFooter = props => {
  const { children, className, ...rest } = props;
  const modalFooterClass = cx('modal-footer', className);
  return (
    <div className={modalFooterClass} {...rest}>
      {children}
    </div>
  );
};

const ModalTitle = props => {
  const { children, className, ...rest } = props;
  const modalTitleClass = cx('modal-title', className);
  return (
    <h3 className={modalTitleClass} {...rest}>
      {children}
    </h3>
  );
};

const ModalClose = props => {
  const { className, ...rest } = props;
  const modalCloseClass = cx('close', className);
  return (
    <button
      type="button"
      className={modalCloseClass}
      data-dismiss="modal"
      aria-label="Close"
      {...rest}
    >
      <span aria-hidden="true">&times;</span>
    </button>
  );
};

Modal.Content = ModalContent;
Modal.Header = ModalHeader;
Modal.Body = ModalBody;
Modal.Footer = ModalFooter;
Modal.Title = ModalTitle;
Modal.Close = ModalClose;
Modal.Trigger = ModalTrigger;

export { Modal, ModalHeader, ModalTitle, ModalBody, ModalFooter, ModalClose, ModalTrigger };

Modal.propTypes = {
  element: PropTypes.string.isRequired,
};

ModalTrigger.propTypes = {
  element: PropTypes.string.isRequired,
};
