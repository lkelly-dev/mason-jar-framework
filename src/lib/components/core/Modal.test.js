import React from 'react';
import { shallow } from 'enzyme';
import { Modal, ModalHeader, ModalTitle, ModalBody, ModalFooter, ModalClose, ModalTrigger } from "./Modal";

describe('Modal', () => {
  it('Modal matches snapshot', () => {
    //FIXME element should not be required. also the name is not great
    const component = shallow(<Modal element="div"></Modal>);
    expect(component).toMatchSnapshot();
  });

  it('has expected props', () => {
    const id = "some-id"
    const children = <div>child</div>
    const component = shallow(
      <Modal element={ id }>
        { children }
      </Modal>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(component.contains(children)).toEqual(true);
    expect(classNameArray).toContain('modal');
    expect(component.prop('id')).toBe(id);
    expect(classNameArray).toHaveLength(1);
  });
});
