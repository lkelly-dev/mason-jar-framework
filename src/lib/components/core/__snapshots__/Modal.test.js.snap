// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Modal matches snapshot 1`] = `
<div
  aria-hidden="true"
  className="modal"
  id="div"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  />
</div>
`;
